src/services/IngestorService.ts

```diff
@@ -19,18 +19,26 @@ export class IngestorService {
                 console.log("error while connecting to kafka", error.message)
             })
     }
-    public create = (req: Request, res: Response, next: NextFunction) => {
-        req.body = Object.assign(req.body.data, { "dataset": this.getDatasetId(req) });
-        this.kafkaConnector.execute(req, res)
-            .then(() => {
-                ResponseHandler.successResponse(req, res, { status: 200, data: { "message": constants.DATASET.CREATED } })
-            }).catch((error: any) => {
-                next({ statusCode: error.status || httpStatus.INTERNAL_SERVER_ERROR, message: error.message || "", errCode: error.code || httpStatus["500_NAME"] })
-            })
+    public create = async (req: Request, res: Response, next: NextFunction) => {
+        try {
+            const datasetId = this.getDatasetId(req);
+            req.body = { ...req.body.data, dataset: datasetId };
+            const topic = await this.getTopic(datasetId);
```

Comment:
Can we have some validation on the topic? What if we receive the topic as empty string ?